yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -976,6 +976,11 @@ private[spark] class Client(
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
           case NonFatal(e) =>
+            if (e.isInstanceOf[InterruptedException]
```

Comment:
we can only move InterruptedException to above but not exception cause by it.
`just move InterruptedException`  or `leave these two here`, which option do you think is better?